### PR TITLE
LGA 1825 p4 child 1 calendar view callback slots

### PR DESCRIPTION
**Pull request by jogilder**
This needs the creation of the fixture containing the cases so can use those references to create the callbacks to be merged first.

Requires updated version of cla_common that can be used with python3. Issue with version of black when looking at pre-commits with that change.

### CODE REVIEW

File: behave/features/steps/cla_pick_up_digital_case_by_call_back.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    # Choose 6 days so can be sure that will not get three bank holidays/sunday etc
+    operator_hours = OpeningHours(**OPERATOR_HOURS)
+    available_days_from_common = operator_hours.available_days(6)
+    all_available_slots = []
+    for day in available_days_from_common:
+        available_slots = operator_hours.time_slots(day.date())
+        all_available_slots.extend(available_slots)
+    # do every other slot as on the display they lump two slots together on website
+    slots_chosen = all_available_slots[:8:2]
+    slots_chosen.append(all_available_slots[0])
```
Comment (said-moj) [inline behave/features/steps/cla_pick_up_digital_case_by_call_back.py:28]: What do you think of putting all these in another function as it's quite complex, I think maybe moving it to another place would make the current function easier to read as you have to keep less details in your head when reading it

File: behave/features/cla_pick_up_digital_case_by_call_back.feature [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+#As a CHS Operator, I want to select a callback slot, so I can see the cases where a callback is booked for that slot.
+Background: Create Callbacks
+    Given that I am logged in
+    And that I have created cases with callbacks
+
+
+@check_callback_exists
+Scenario: Can see callbacks booked in different slots
+    Given that I am on cases callback page located at /call_centre/callbacks/
```
Comment (said-moj) [inline behave/features/cla_pick_up_digital_case_by_call_back.feature:12]: What do you think of maybe replacing this with `Given that I am on cases callback page`?

File: behave/features/steps/cla_pick_up_digital_case_by_call_back.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    start_page_url = f"{CLA_FRONTEND_URL}/call_centre/callbacks/"
+    context.helperfunc.open(start_page_url)
+    assert_header_on_page("Cases", context)
+
+
+@given(u'multiple cases with a callback exists')
+def step_impl(context):
+    #  there will be at least one <a> element that shows callbacks are booked
+    callbacks_exist = context.helperfunc.find_many_by_class("CallbackMatrix-slot")
+    assert len(callbacks_exist) > 1, f"Only {len(callbacks_exist)} callback, want more than one "
```
Comment (said-moj) [inline behave/features/steps/cla_pick_up_digital_case_by_call_back.py:73]: Can the error message read something like: Found only x callbacks, expected to have more than one callback

File: behave/features/steps/cla_pick_up_digital_case_by_call_back.py (hunk 1 above)
Comment (jogilder) [reply behave/features/steps/cla_pick_up_digital_case_by_call_back.py:28]: will do

File: behave/features/steps/cla_pick_up_digital_case_by_call_back.py (hunk 3 above)
Comment (jogilder) [reply behave/features/steps/cla_pick_up_digital_case_by_call_back.py:73]: k

**Review by said-moj — APPROVED** [review]
👍  LGTM

**Pull request merged**